dotnet/src/webdriver/Internal/AsyncHelper.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="AsyncHelper.cs" company="WebDriver Committers">
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenQA.Selenium.Internal
+{
+    /// <summary>
+    /// Encapsulates methods for working with asynchronous tasks.
+    /// </summary>
+    public static class AsyncHelper
+    {
+        private static readonly TaskFactory _myTaskFactory = new TaskFactory(CancellationToken.None,
+            TaskCreationOptions.None, TaskContinuationOptions.None, TaskScheduler.Default);
+
+        public static TResult RunSync<TResult>(Func<Task<TResult>> func)
+        {
+            return _myTaskFactory.StartNew(() =>
+            {
+                return func();
+            }).Unwrap().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
`ConfigureAwait(false)` seems to have no sense here and in the method below, according to the statement from [ConfigureAwait in .NET 8](https://blog.stephencleary.com/2023/11/configureawait-in-net-8.html) article:

> `ConfigureAwait` configures the `await`, not the task. E.g., the `ConfigureAwait(false)` in `SomethingAsync().ConfigureAwait(false).GetAwaiter().GetResult()` does exactly nothing. Similarly, the `await` in `var task = SomethingAsync(); task.ConfigureAwait(false); await task;` still continues on the captured context, completely ignoring the `ConfigureAwait(false)`.